js/main.js

```diff
@@ -208,64 +208,46 @@ function StravaEnhancementSuite($, options) {
 
   // Methods //////////////////////////////////////////////////////////////////
 
-  // Own activity pages
+  // Own activity page
   $.always(function() {
     // Detect whether we are viewing our own activity and to activate the dialog
-    var edit_activity = $('.sidenav .edit-activity');
-
-    // Seems to be truncated in size
-    $('#heading span.title')
-      .css('display', 'inline')
-      ;
+    var edit_activity = $('a[title="Edit this activity"]');
 
     if (edit_activity.length === 0) {
       return;
     }
 
-    // Clicking the activity name launches the dialog
-    $('#heading h2.activity-name')
+    // Edit activity by clicking on it's name
+    $('.activity-name')
       .css('cursor', 'pointer')
       .on('click', function() {
-        edit_activity.click();
+        location.pathname = edit_activity.attr('href')
```

Comment:
edit_activity is now plain <a> with no onClick handler, just plain href attribute. Calling .click() on it did not work for me, so I've rewritten it this way. But I'm by no means jQuery expert